stdlib/public/core/Array.swift

```diff
@@ -803,6 +803,7 @@ extension Array: RangeReplaceableCollection {
   ///     // Prints "true"
   @inlinable
   @_semantics("array.init")
+  @_semantics("array.init_empty")
   public init() {
```

Comment:
I wonder if we need a new semantics attribute here or instead check that the parameters of the initializer are empty  and that the receiver object is Array, and return unknown if it is not.

@eeckstein This PR is adding support for Arrays to the interpreter. We would like to handle only `Array.init()`, and will have use a semantics annotation to hardcode its semantics in the interpreter. There seems to be an "array.init" semantics attribute on Array initializers, but it is shared by empty and non-empty initializes as well as by other types: `ArraySilce` and `ContiguousArray`. So, to distinguish `Array.init()` there are two options, either we add another semantics attribute to `Array.init()`, or check the receiver type and the parameter count in the interpreter to identify this initializer from others. Just wondering what approach you would recommend here?